PURL-TYPES.rst

```diff
@@ -81,15 +81,24 @@ conan
 -----
 ``conan`` for Conan C/C++ packages:
 
-- The default repository is ``https://center.conan.io``
-- The ``namespace`` is the user if present
-- The ``name`` is the package name.
-- The ``version`` is the package version.
-- The qualifier ``channel`` must be not empty if namespace is present
+- The default repository is ``https://conan.io/center``.
+- The ``namespace`` is the vendor.
+- The ``name`` is the conan package name.
+- The ``version`` is the conan package version.
+- The qualifier ``channel`` is optionally the conan channel.
+- The qualifier ``user`` is optionally the conan user.
+- The qualifier ``rrev`` is optionally the conan recipe revision. If none is given, the latest recipe revision available for the given version is meant.
```

Comment:
"If none is given" could be interpreted as literally passing "rrev=none". Maybe better say "If omitted, ...".

Also, maybe better "is used" instead of "is meant"?

Similar below.

Reply:
about None: good point
about "is used" vs. "is meant" i am not so sure. The case where no rrev is specified but multiple revisions exist is a bit strange because in the end the user would need to look up what the newest one is. "is used" implies that some implementation exists which would just do that, but this is not the case.
So what i want to convey here that writing a PURL without rrev/prev when multiple revisions exists shall be interpreted as "latest revision to be determined by the user".

Reply:
Where "latest revision to be determined by the user" could be *any* of the available revisions, and not necessarily the one with the latest date or so? Then that's rather arbitrary, and maybe we should simply drop the second sentence completely, implying that it's undefined (similar to how we also don't explicitly say what omitting any of the other optional qualifiers means).

Reply:
Conan behaves the way that if you dont specificy prev or rrev, then it will fetch the latest one from the server. I think the same behaviour we should also strive for. Just here in the PURL-spec document we dont have a server and we dont have an implemented tool, so it is just an intention we can formulate.

Reply:
> I think the same behaviour we should also strive for.

Agreed, but I'm still wondering why we're documenting the implicit default behavior here, but not for the other optional qualifiers? I.e. which channel is used if none is specified?

Reply:
Hmm, @sschuberth user and channel are also optional in Conan, so i think having them optional here as well kind of makes sense. Shall we define that they shall only be omitted if also on the conan package side they were omitted?

Reply:
> Shall we define that they shall only be omitted if also on the conan package side they were omitted?

With my Conan-half-knowledge, I'd say that makes sense, yes.

Reply:
The structure differs now from the other package types, but i think it is more clear and consistent in itself now.